StitchCore/StitchCore/Source/Core/StitchClient.swift

```diff
@@ -268,27 +276,41 @@ public class StitchClient: StitchClientType {
     public func login(withProvider provider: AuthProvider) -> Promise<UserId> {
         self.authProvider = provider
 
-        if isAuthenticated, let auth = auth {
-            printLog(.info, text: "Already logged in, using cached token.")
-            return Promise.init(value: auth.userId)
+        func doLoginRequest() -> Promise<UserId> {
+            return httpClient.doRequest {
```

Comment:
[style] This is more for my education, but how much do we personally like the usage of `${0..n}` as opposed to just specifying the parameter in the function closure?

Reply:
I'm usually for explicitly specifying parameters for the sake of readability but I'll let @jsflax decide since he's more familiar with what's normal in Swift.